cmd/blockstorage_detach.go

```diff
@@ -0,0 +1,79 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/exoscale/cli/pkg/globalstate"
+	"github.com/exoscale/cli/pkg/output"
+	v3 "github.com/exoscale/egoscale/v3"
+)
+
+type blockstorageDetachCmd struct {
+	cliCommandSettings `cli-cmd:"-"`
+
+	_ bool `cli-cmd:"detach"`
+
+	Volume string      `cli-arg:"#" cli-usage:"NAME|ID"`
+	Force  bool        `cli-short:"f" cli-usage:"don't prompt for confirmation"`
+	Zone   v3.ZoneName `cli-short:"z" cli-usage:"block storage zone"`
+}
+
+func (c *blockstorageDetachCmd) cmdAliases() []string { return gCreateAlias }
+
+func (c *blockstorageDetachCmd) cmdShort() string { return "Detach a Block Storage Volume" }
+
+func (c *blockstorageDetachCmd) cmdLong() string {
+	return fmt.Sprintf(`This command detaches Block Storage Volume.
+
+Supported output template annotations: %s`,
+		strings.Join(output.TemplateAnnotations(&blockstorageShowOutput{}), ", "))
+}
+
+func (c *blockstorageDetachCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
+	cmdSetZoneFlagFromDefault(cmd)
+	return cliCommandDefaultPreRun(c, cmd, args)
+}
+
+func (c *blockstorageDetachCmd) cmdRun(_ *cobra.Command, _ []string) error {
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
+	}
+
+	volumes, err := client.ListBlockStorageVolumes(ctx)
+	if err != nil {
+		return err
+	}
+
+	volume, err := volumes.FindBlockStorageVolume(c.Volume)
+	if err != nil {
+		return err
+	}
```

Comment:
i've seen this several times now, we could put it in a helper

Reply:
It's short enough and totally readable to leave it like that, put it in a helper will make the readiness longer/complicated